source/material_model/viscoelastic_plastic.cc

```diff
@@ -0,0 +1,737 @@
+          if (dim == 2)
+            {
+              AssertThrow(this->introspection().compositional_index_for_name("stress_xx") == 0,
+                          ExcMessage("Material model Viscoelastic only works if the first "
+                                     "compositional field is called stress_xx."));
+              AssertThrow(this->introspection().compositional_index_for_name("stress_yy") == 1,
+                          ExcMessage("Material model Viscoelastic only works if the second "
+                                     "compositional field is called stress_yy."));
+              AssertThrow(this->introspection().compositional_index_for_name("stress_xy") == 2,
+                          ExcMessage("Material model Viscoelastic only works if the third "
+                                     "compositional field is called stress_xy."));
+            }
+          else if (dim == 3)
+            {
+              AssertThrow(this->introspection().compositional_index_for_name("stress_xx") == 0,
+                          ExcMessage("Material model Viscoelastic Plastic only works if the first "
+                                     "compositional field is called stress_xx."));
+              AssertThrow(this->introspection().compositional_index_for_name("stress_yy") == 1,
+                          ExcMessage("Material model Viscoelastic Plastic only works if the second "
+                                     "compositional field is called stress_yy."));
+              AssertThrow(this->introspection().compositional_index_for_name("stress_zz") == 2,
+                          ExcMessage("Material model Viscoelastic Plastic only works if the third "
+                                     "compositional field is called stress_zz."));
+              AssertThrow(this->introspection().compositional_index_for_name("stress_xy") == 3,
+                          ExcMessage("Material model Viscoelastic Plastic only works if the fourth "
+                                     "compositional field is called stress_xy."));
+              AssertThrow(this->introspection().compositional_index_for_name("stress_xz") == 4,
+                          ExcMessage("Material model Viscoelastic Plastic only works if the fifth "
+                                     "compositional field is called stress_xz."));
+              AssertThrow(this->introspection().compositional_index_for_name("stress_yz") == 5,
+                          ExcMessage("Material model Viscoelastic Plastic only works if the sixth "
+                                     "compositional field is called stress_yz."));
+            }
+          else
+            ExcNotImplemented();
+
+          // It does not make sense to use this material with the 'no Advection, iterated Stokes'
+          // 'single Advection, no Stokes', 'iterated Advection and Stokes' or 'iterated Advection
+          // and Newton Stokes' solver schemes.
```

Comment:
I think you can remove this comment as it just states what is described in the error message anyway. Or document *why* it is only possible with the allowed solver schemes.